Index: libs/submit.php
===================================================================
--- libs/submit.php	(revision 16588)
+++ libs/submit.php	(working copy)
@@ -1530,8 +1530,8 @@
 				$Request = array_merge((($type == 'Account') ? array('i_account'=>$submit_info['i_account']) : array('i_customer'=>$submit_info['i_customer'])),
 					array(
 						'i_service' => 2,
-						'from_date' => date('m-d-Y h:i:s',strtotime('-1 day')),
-						'to_date'=>'now'
+						'from_date' => date('Y-m-d H:i:s',strtotime('-1 day')),
+						'to_date'=>date('Y-m-d H:i:s')
 					)
 				);
 				$method = ($type == 'Account') ? 'get_xdr_list' : 'get_customer_xdrs';
