Index: libs/submit.php
===================================================================
--- libs/submit.php	(revision 16708)
+++ libs/submit.php	(working copy)
@@ -248,7 +248,7 @@
 
 					{
 						$env = '0'.$env;
 					}
-					$qrcode = 'csc:'.$env.$GLOBALS['account_info']['id'].':'.$GLOBALS['account_info']['h323_password'].'@TG';
+					$qrcode = 'csc:'.$GLOBALS['account_info']['id'].':'.$GLOBALS['account_info']['h323_password'].'@TG';
 				}
 				$package_desc = empty($set_package[$GLOBALS['lang']]['description']) ? ( $pack = reset($set_package) and !empty($pack['description']) ? $pack['description'] : '' ) : $set_package[$GLOBALS['lang']]['description'];
 				if(isset($GLOBALS["virtoffice"])) $subscriber["virtoffice"] = TRUE;
