Index: libs/submit.php
===================================================================
--- libs/submit.php	(revision 16776)
+++ libs/submit.php	(working copy)
@@ -807,7 +807,7 @@
 
 		foreach ($GetCustomerServiceFeaturesResponse->service_features as $value)
 		{
-                    if (in_array($value->name, array("rtpp_level", "voice_dialing", "unified_messaging"))) {
+                    if (in_array($value->name, array("rtpp_level", "voice_dialing"))) {
                     $ArrayCustomerServiceFeatures[$value->name] = $value;
                     }
                 }
@@ -1073,7 +1073,7 @@
 
 		foreach ($GetAccountServiceFeaturesResponse->service_features as $value)
 		{
-                    if (in_array($value->name, array("rtpp_level", "voice_dialing", "routing_plan", "unified_messaging", "auto_attendant"))) {
+                    if (in_array($value->name, array("rtpp_level", "voice_dialing", "routing_plan", "unified_messaging", "auto_attendant", "music_on_hold"))) {
                         $ArrayAccountServiceFeatures[$value->name] = $value;
                         }
                 }
